### PR TITLE
AP_Scripting/Copter: support is_landing and is_taking_off in lua script

### PR DESCRIPTION
**Pull request by tatsuy**
I added is_landing and is_taking_off function to the AP_Vehicle and Copter class to use these in LUA scripting.
These can be used, for example, to specify failsafe behavior during takeoff and landing at LUA.

I tested this in SITL and it works.

### CODE REVIEW

File: ArduCopter/Copter.h [hunk 1]
```diff
@@ -679,6 +679,8 @@ class Copter : public AP_Vehicle {
     // lua scripts use this to retrieve EKF failsafe state
     // returns true if the EKF failsafe has triggered
     bool has_ekf_failsafed() const override;
+    bool is_landing() override;
```
Comment (rmackay9) [inline ArduCopter/Copter.h:682]: Looks pretty good.  These could be made const I think.

Comment (tatsuy) [reply ArduCopter/Copter.h]: Thank you! I fixed it.

**Review by rmackay9 — APPROVED** [review]
LGTM

**Comment by rmackay9** [comment]
By the way, there is apparently already a get_likely_flying() binding.

**Comment by tatsuy** [comment]
@rmackay9 Good!
I will add landed_state instead of is_landing and is_taking_off.

**Comment by tatsuy** [comment]
@rmackay9 
I have created a modification to use landed_state.
https://github.com/ArduPilot/ardupilot/compare/master...tatsuy:ardupilot:pr-landed-state?expand=1

Both "landed_state" and "vtol_state" included in EXTENDED_SYS_STATE can be obtained from lua.
But, changes will be made to Copter, Plane and Blimp.
Which change is better?

**Comment by muramura** [comment]
I think if you are going to change the ArduPilot's internal information to the MAVLINK definition, you should leave GCS_MAVLINK.CPP as it is.

**Review by IamPete1 — DISMISSED** [review]
Needs lua docs:
```
function vehicle:is_taking_off() end
function vehicle:is_landing() end
```

**Comment by tatsuy** [comment]
@IamPete1 Thank you!
I added lua docs.

**Comment by rmackay9** [comment]
Two approvals so in it goes once it passes CI. txs!

**Pull request merged**